src/Symfony/Component/Console/Input/Input.php

```diff
@@ -72,7 +72,12 @@ public function bind(InputDefinition $definition)
      */
     public function validate()
     {
-        if (count($this->arguments) < $this->definition->getArgumentRequiredCount()) {
+        $definition = $this->definition;
+        $requiredArguments = array_filter(array_keys($this->arguments), function ($argument) use ($definition) {
+            return $definition->hasArgument($argument) && $definition->getArgument($argument)->isRequired();
+        });
+
+        if (count($requiredArguments) < $this->definition->getArgumentRequiredCount()) {
```

Comment:
`if (count($requiredArguments) < $definition->getArgumentRequiredCount()) {` would be shorter